src/onboarding/restore-backup/onboarding-restore-backup-cloud-login.ts

```diff
@@ -0,0 +1,225 @@
+import "../../components/ha-icon-button-arrow-prev";
+import type { LocalizeFunc } from "../../common/translations/localize";
+import type { BackupContentExtended } from "../../data/backup";
+import { fireEvent } from "../../common/dom/fire_event";
+import { forgotPasswordHaCloud, loginHaCloud } from "../../data/onboarding";
+import { handleCloudLoginError } from "../../data/cloud";
+import { navigate } from "../../common/navigate";
+import { removeSearchParam } from "../../common/url/search-params";
+import { onBoardingStyles } from "../styles";
+
+@customElement("onboarding-restore-backup-cloud-login")
+class OnboardingRestoreBackupCloudLogin extends LitElement {
+  @property({ attribute: false }) public localize!: LocalizeFunc;
+
+  @property({ attribute: false }) public backup!: BackupContentExtended;
+
+  @state() private _requestInProgress = false;
+
+  @state() private _email?: string;
+
+  @state() private _password?: string;
+
+  @state() private _error?: string;
+
+  @state() private _view: "login" | "forgot-password" | "loading" = "login";
+
+  @state() private _showResetPasswordDone = false;
+
+  render() {
+    return html`
+      <ha-icon-button-arrow-prev
+        .label=${this.localize("ui.panel.page-onboarding.restore.back")}
+        @click=${this._back}
+      ></ha-icon-button-arrow-prev>
+      <h1>Home Assistant Cloud</h1>
+      <p>
+        ${this.localize(
+          "ui.panel.page-onboarding.restore.ha-cloud.sign_in_description"
+        )}
+      </p>
+      ${this._showResetPasswordDone ? this._renderResetPasswordDone() : nothing}
+      ${this._view === "login"
+        ? html`<cloud-login
+            card-less
+            .email=${this._email}
+            .password=${this._password}
+            .inProgress=${this._requestInProgress}
+            .error=${this._error}
+            .localize=${this.localize}
+            translation-key-panel="page-onboarding.restore.ha-cloud"
+            @cloud-login=${this._handleLogin}
+            @cloud-forgot-password=${this._showForgotPassword}
+          ></cloud-login>`
+        : this._view === "loading"
+          ? html`<div class="loading">
+              <ha-circular-progress
+                indeterminate
+                size="large"
+              ></ha-circular-progress>
+            </div>`
+          : html`<cloud-forgot-password-card
+              card-less
+              .localize=${this.localize}
+              .inProgress=${this._requestInProgress}
+              .error=${this._error}
+              translation-key-panel="page-onboarding.restore.ha-cloud.forgot_password"
+              @cloud-forgot-password=${this._handleForgotPassword}
+            ></cloud-forgot-password-card>`}
+    `;
+  }
+
+  private _back() {
+    if (this._view === "forgot-password") {
+      this._view = "login";
+      return;
+    }
+
+    navigate(`${location.pathname}?${removeSearchParam("page")}`);
+  }
+
+  private _doLogin = async (
```

Comment:
Why isn't the `cloud-login` element handling this? Because we have 2 different endpoints? The rest of the implementation seems similar? As we use the same error handling function. I feel like that should move to `cloud-login`, would make it cleaner and simpeler.

Reply:
Yes that makes sense, it's just the cloud_pipeline confirmation, but we can add a prop for that.